### PR TITLE
(DAL) Implement basic api key

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

Basic api key implementation for beta

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [x] Should publish Docker image

### CODE REVIEW

**Comment by nick-bisonai** [comment]
@coderabbitai review

File: node/pkg/dal/utils/utils.go [hunk 1]
```diff
@@ -17,6 +17,23 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
+func APIKeyMiddleware() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		apiKey := c.Get("X-API-Key")
+
+		validAPIKey := os.Getenv("API_KEY")
+
+		if apiKey != validAPIKey {
```
Comment (Intizar-T) [inline node/pkg/dal/utils/utils.go:26]: just out of curiosity, does this mean we'll have a single api key for all users?

Comment (nick-bisonai) [reply node/pkg/dal/utils/utils.go:26]: yh it's just beta version, not for final milestone proof

**Review by Intizar-T — APPROVED** [review]
lgtm!

**Pull request merged**